sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2963,7 +2963,11 @@ class Analyzer(override val catalogManager: CatalogManager)
 
     private def getNondeterToAttr(exprs: Seq[Expression]): Map[Expression, NamedExpression] = {
       exprs.filterNot(_.deterministic).flatMap { expr =>
-        val leafNondeterministic = expr.collect { case n: Nondeterministic => n }
+        val leafNondeterministic = expr.collect {
+          case n: Nondeterministic => n
+          case e: CallMethodViaReflection => e
```

Comment:
This fix itself looks fine. Just my nit comment: naming `java_method('java.lang.Math', 'abs', c1)` "_nondeterministic " in a plan looks a bit weird.

Reply:
Just for simple test, like the empty scala udf.

Reply:
Ah, ok. I misunderstood it a bit. nvm.